.github/scripts/trymerge.py

```diff
@@ -1107,11 +1107,14 @@ def delete_land_time_check_branch(self,
         repo._run_git('push', 'origin', '-d', land_check_branch)
 
 
-class MandatoryChecksMissingError(Exception):
+class MergeRuleFailedError(RuntimeError):
```

Comment:
Just curious q about adding `MergeRuleFailedError` base class.  Better naming? or is this anticipating more type of error?

Reply:
I could've simply added a new exception without turning it into a base class, but this seemed to logically be a more general case of the pre-existing `MandatoryChecksMissingError`.  I couldn't merge the two into one Exception since `MandatoryChecksMissingError` needs to be caught and reacted to differently in some cases